src/services/utilities.ts

```diff
@@ -927,4 +933,101 @@ namespace ts {
         }
         return ensureScriptKind(fileName, scriptKind);
     }
+
+    export function enumerateNodeModulesVisibleToScript(host: LanguageServiceHost, scriptPath: string, modulePrefix?: string) {

```

Comment:
WOW.. this is a lot of IO just to get a completion entry. we do not have that much budget. if each package takes 1ms, and you have 200 packages in your node_modules, this is 200ms, we still need to create a checker, bind the whole program...

we should talk about this part. we should look into possible caching opportunities given that the package.json file is likely not change between invocations.

i would recommend not including this in the first iteration.